src/solvers/refinement/string_constraint_generator_concat.cpp

```diff
@@ -11,44 +11,60 @@ Author: Romain Brenguier, romain.brenguier@diffblue.com
 
 /*******************************************************************\
 
-Function: string_constraint_generatort::add_axioms_for_concat
+Function: string_constraint_generatort::add_axioms_for_concat_substr
 
-  Inputs: two string expressions
+  Inputs:
+    s1 - string expression
+    s2 - string expression
+    start_index - expression representing an integer
+    end_index - expression representing an integer
 
  Outputs: a new string expression
 
  Purpose: add axioms to say that the returned string expression is equal to
-          the concatenation of the two string expressions given as input
+          the concatenation of s1 with the substring of s2 starting at index
+          start_index and ending at index end_index.
+          If start_index >= end_index the value returned is s1.
+
+          TODO: For now it just assumes end_index < s2.length. This
```

Comment:
@romainbrenguier is right. In the present situation, the constraints do not really allow to for the case where the indices are out of bound, which is fine for the general case, but will not work when we try to throw an exception from the model. Although it is not quite clear what should happen with exceptions, I will relax the constraints for the out of bound case (which is more or less opposite to what was asked for). :-)